api/indexer.oas2.json

```diff
@@ -2677,6 +2738,61 @@
     }
   },
   "parameters": {
+    "proposer": {
+      "type": "array",
+      "items": {
+        "type": "string",
+        "x-algorand-format": "Address"
+      },
+      "description": "Account(s) marked as proposer in the block header's participation updates. This parameter accepts a comma separated list of addresses.",
+      "name": "proposer",
+      "in": "query",
+      "required": false
+    },
+    "absent": {
+      "type": "array",
+      "items": {
+        "type": "string",
+        "x-algorand-format": "Address"
+      },
+      "description": "Account(s) marked as absent in the block header's participation updates. This parameter accepts a comma separated list of addresses.",
+      "name": "absent",
+      "in": "query",
+      "required": false
+    },
+    "expired": {
+      "type": "array",
+      "items": {
+        "type": "string",
+        "x-algorand-format": "Address"
+      },
+      "description": "Account(s) marked as expired in the block header's participation updates. This parameter accepts a comma separated list of addresses.",
+      "name": "expired",
+      "in": "query",
+      "required": false
+    },
+    "updates": {
+      "type": "array",
+      "items": {
+        "type": "string",
+        "x-algorand-format": "Address"
+      },
+      "description": "Account(s) marked as expired or absent in the block header's participation updates. This parameter accepts a comma separated list of addresses.",
+      "name": "updates",
+      "in": "query",
+      "required": false
+    },
+    "participation": {
+      "type": "array",
+      "items": {
+        "type": "string",
+        "x-algorand-format": "Address"
+      },
+      "description": "Account(s) marked as expired, absent or as proposer in the block header's participation updates. This parameter accepts a comma separated list of addresses.",
```

Comment:
This seems like a weird catch-all. I guess the intent is for block explorers that want to show any relevant block for particular accounts?

Reply:
I believe so. Maybe @urtho can give more context on how the `participation` parameter is intended to be used.

Performance-wise, it is the most problematic parameter. I can see the standard version of the indexer not including the `participation` parameter (I regard it as a low benefit / high cost feature)

Something similar happens with the `updates` parameter, maybe not as bad.

Reply:
> Is there any meaningful advantage to it, compared to calling the endpoint three times, specifying the address for each of the three things? I would guess you're using an index for each, so it's basically three index lookups no matter how you slice it.
> 
> These lists (expired and suspended) are almost always empty. It seems like an index lookup would be super fast. Just do two of them?

That's exactly right. The essential functionality is already being exposed by the "atomic" parameters (that is: `proposer`, `absent` and `expired`)

The data returned by the `updates` and `participation` parameters could be obtained by calling the endpoint multiple times using the "atomic" parameters.

Reply:
Addressed in https://github.com/algorand/indexer/pull/1638/commits/8cb90c11876a9bc8fa70f42c0b7321a6625bdfba